modulus/datapipes/climate/climate_hdf5.py

```diff
@@ -0,0 +1,590 @@
+            use_latlon = True
+        self.use_latlon = use_latlon
+        self.use_cos_zenith = use_cos_zenith
+        self.process_rank = process_rank
+        self.world_size = world_size
+        if isinstance(patch_size, int):
+            patch_size = (patch_size, patch_size)
+        self.patch_size = patch_size
+        self.num_samples_per_year = num_samples_per_year
+
+        # Determine outputs of pipeline
+        self.pipe_outputs = ["state_seq", "timestamps"]
+        if self.lsm_filename is not None:
+            self.pipe_outputs.append("land_sea_mask")
+        if self.geopotential_filename is not None:
+            self.pipe_outputs.append("geopotential")
+        if self.use_latlon:
+            self.pipe_outputs.append("latlon")
+            self.pipe_outputs.append("cos_latlon")
+        if self.use_cos_zenith:
+            self.pipe_outputs.append("cos_zenith")
+
+        # Set up device, needed for pipeline
+        if isinstance(device, str):
+            device = torch.device(device)
+
+        # Need a index id if cuda
+        if device.type == "cuda" and device.index == None:
+            device = torch.device("cuda:0")
+        self.device = device
+
+        # check root directory exists
+        if not self.data_dir.is_dir():
+            raise IOError(f"Error, data directory {self.data_dir} does not exist")
+        if not self.stats_dir is None and not self.stats_dir.is_dir():
+            raise IOError(f"Error, stats directory {self.stats_dir} does not exist")
+
+        # Load all data files and statistics
+        self._parse_dataset_files()
+        self._load_statistics()
+        if self.lsm_filename is not None:
+            self._load_land_sea_mask()
+        if self.geopotential_filename is not None:
+            self._load_geopotential()
+        if self.use_latlon:
+            self._load_latlon()
+
+        # Create pipeline
+        self.pipe = self._create_pipeline()
+
+    def _parse_dataset_files(self) -> None:
+        """Parses the data directory for valid HDF5 files and determines training samples
+
+        Raises
+        ------
+        ValueError
+            In channels specified or number of samples per year is not valid
+        """
+        # get all input data files
+        self.data_paths = sorted(self.data_dir.glob("*.h5"))
+        for data_path in self.data_paths:
+            self.logger.info(f"Climate file found: {data_path}")
+        self.n_years = len(self.data_paths)
+        self.logger.info(f"Number of years: {self.n_years}")
+
+        # get total number of examples and image shape from the first file,
+        # assuming other files have exactly the same format.
+        self.logger.info(f"Getting file stats from {self.data_paths[0]}")
+        with h5py.File(self.data_paths[0], "r") as f:
+            # truncate the dataset to avoid out-of-range sampling
+            data_samples_per_year = f["fields"].shape[0] - self.num_steps * self.stride
```

Comment:
Shouldn't this be  `data_samples_per_year = f["fields"].shape[0] - (self.num_steps-1) * self.stride`

Reply:
Fixed this.